src/averageAges.js

```diff
@@ -36,9 +47,24 @@ function calculateMenAverageAge(people, century) {
  *
  * @return {number}
  */
+
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = people.filter(person => person.sex === 'f');
+
+  const allMothers = people.map(person => person.mother);
+
+  const womenWithChild = withChildren
+    ? women.filter(mother => allMothers.includes(mother.name))
+    : women;
+
+  const lifeTime = womenWithChild.map(woman => woman.died - woman.born);
+
+  const averageWomanLife = (lifeTime.reduce((life1, life2) =>
+    life1 + life2)) / womenWithChild.length;
+
+  return averageWomanLife;
```

Comment:
you repeat yourself here, you basically do the same logic in `men` function, create function helpers for it